src/cdk/a11y/live-announcer/live-announcer.ts

```diff
@@ -149,17 +160,57 @@ export class LiveAnnouncer implements OnDestroy {
     }
   }
 
-  private _createLiveElement(): HTMLElement {
-    const elementClass = 'cdk-live-announcer-element';
-    const previousElements = this._document.getElementsByClassName(elementClass);
-    const liveEl = this._document.createElement('div');
+  private _enqueueAnnouncement(
+      message: string, politeness: AriaLivePoliteness, duration?: number): Promise<void> {
+    const liveElement = this._createLiveElement();
+    // TODO: ensure changing the politeness works on all environments we support.
+    liveElement.setAttribute('aria-live', politeness);
+
+    return new Promise(resolve => {
+      this._taskQueue.enqueue(() => {
+        liveElement.textContent = message;
+        resolve();
+
+        duration = duration ? duration : ARIA_LIVE_ANNOUNCER_REMOVAL_DELAY_MS;
+        setTimeout(() => {
+          document.body.removeChild(liveElement);
```

Comment:
This will throw during server-side rendering.

Reply:
I've replaced `document` with `this._document`